From 3272ae24ac21867b23937f879ea2e43e456fbece Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 7 Feb 2013 16:03:51 +0800
Subject: [PORT FROM MAIN]libmediaplayerservice: shutdown video decoder when it has error and wait discontinuity

BZ: 81083

when video decoder's loop stops after video omx client reports error
if it needs discontinuity flag then, it can not get it. Shut down video decoder
in this condition to avoid reset ANR.

Change-Id: Id868e3140b0f9bb476c435f57009103bfe59c982
Orig-Change-Id: Ie60d924d146270a43d6b33079b146c4deb98cdfd
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libmediaplayerservice/nuplayer/NuPlayer.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index 00ce1db..c2ec6fd 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -707,6 +707,12 @@ void NuPlayer::onMessageReceived(const sp<AMessage> &msg) {
                     ALOGE("%s track encountered an error (%d)",
                          audio ? "audio" : "video", finalResult);
 
+                    if (!audio && (mVideoDecoder != NULL)
+                            && (mFlushingVideo == AWAITING_DISCONTINUITY)) {
+                        // if video decoder is in AWAITING_DISCONTINUITY state
+                        // shut down it to avoid reset blocking
+                        flushDecoder(false,true);
+                    }
                     notifyListener(
                             MEDIA_ERROR, MEDIA_ERROR_UNKNOWN, finalResult);
                 }
-- 
1.7.4.1

